Tasks/AzureResourceGroupDeployment/models/DeployAzureRG.ts

```diff
@@ -106,14 +76,72 @@ export class AzureRGTaskParameters {
         }
     }
 
-    private getARMCredentials(connectedService: string): msRestAzure.ApplicationTokenCredentials {
+    private  async getARMCredentials(connectedService: string){
```

Comment:
You should modify the signature to specify the return type i.e. Promise<msrestAzure...>